### PR TITLE
Update cosmosDB state `partitionkey` usage

### LINKED ISSUE #1845: Cosmos DB state store `partitionKey` example is invalid for custom partitions
**Describe the issue**
Cosmos DB state store documentation on specifying [partition key values](https://docs.dapr.io/reference/components-reference/supported-state-stores/setup-azure-cosmosdb/#partition-keys) is incorrect when a user tries to use it for custom defined partition keys (not just the partition key value), and does not accurately convey how the property can be successfully used.

See https://github.com/dapr/cli/issues/805 for details.

**URL of the docs**
https://docs.dapr.io/reference/components-reference/supported-state-stores/setup-azure-cosmosdb/#partition-keys

**Expected content**
The doc should provide a usable example of setting the `partitionKey` value, in relation to how the target cosmosDB container partition key was configured.

See https://github.com/dapr/cli/issues/805 for details.

**Additional context**
There is an additional issue that may be a functional bug, where the default expectation that the `key` property can be used by the state store when `partitionKey` is not specified is effectively unusable because the component always modifies the `key` value to create the `id` in the cosmosDB entry, and then expects the `id` value to be the partition key value which would not occur naturally.

### PR DESCRIPTION
**Pull request by CodeMonkeyLeet**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**

- [x] [Read the contribution guide](https://docs.dapr.io/contributing/contributing-docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [x] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

- Increase emphasis of default `/paritionKey` partition key for CosmosDB collections for use with Dapr state stores.
- Add example of using `partitionKey` metadata with CosmosDB collection using custom partition keys.
- Add `keyPrefix` configuration documentation to state_api.md.

## Issue reference

Fixes #1845

### CODE REVIEW

**Comment by CodeMonkeyLeet** [comment]
@artursouza @berndverst to review for technical accuracy. The behaviors documented are empirically determined, but it's unclear to me whether all of it is intended.

File: daprdocs/content/en/reference/api/state_api.md [hunk 1]
```diff
@@ -403,12 +403,22 @@ spec:
 
 ## Optional behaviors
 
-### Key scheme
```
Comment (artursouza) [inline daprdocs/content/en/reference/api/state_api.md:406]: Keep the title.

**Review by artursouza — CHANGES_REQUESTED** [review]
The Dapr state store API is an opinionated implementation and does not offer a "bring your own schema" functionality. For that, we plan to support a new API for databases, where apps will be able to bring their own schemas and key structure.

https://github.com/dapr/dapr/issues/3354

File: daprdocs/content/en/reference/api/state_api.md (hunk 1 above)
Comment (CodeMonkeyLeet) [reply daprdocs/content/en/reference/api/state_api.md:406]: @artursouza FYI, the key scheme section under Optional Behaviors is redundant as there is a top level `## Key Scheme` section on line 37 that repeats the same info. Given that the key scheme isn't really an optional behavior, it should probably be removed.

**Pull request closed without merging**